components/renku_data_services/connected_services/orm.py

```diff
@@ -77,6 +79,7 @@ class OAuth2ConnectionORM(BaseORM):
     token: Mapped[dict[str, Any] | None] = mapped_column("token", JSONVariant)
     state: Mapped[str | None] = mapped_column("state", String(), index=True, unique=True)
     status: Mapped[ConnectionStatus]
+    extra_data: Mapped[dict[str, Any] | None] = mapped_column("extra_data", JSONVariant)
```

Comment:
I think it is better to not have such general fields in the DB and just have a field specifically for the next_url. Especially when it comes to OAuth. Because it increases the risk that someone would insert something in this dictionary that should not be shared or used in a specific way but we would not know. And the name `extra_data` does not really communicate what the purpose of this data is. 

Also we store the code verifier in this `extra_data` field. But the code verifier is supposed to be used only for a single authentication request and with this setup it could happen that 2 different requests conflict with one another or one request overwrites an existing code verifier of a authentication that is in progress. In a lot of cases the code verifier is just stored in an encypted cookie. And I was hoping that the auth library will handle this for you.

Reply:
Note that `authlib` does not handle client sessions, so it will not save cookies or the like.

Done in a10a435f31628c9df8758ff1fed407bcb5788858.